hack/build.sh

```diff
@@ -22,4 +22,4 @@ HACK_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 source ${HACK_DIR}/env.sh
 
 echo "Building image ${ARGOCD_OPERATOR_IMAGE}"
-${OPERATOR_SDK} build ${ARGOCD_OPERATOR_IMAGE} --image-builder ${ARGOCD_OPERATOR_IMAGE_BUILDER}
+${OPERATOR_SDK} build ${ARGOCD_OPERATOR_IMAGE} ${ARGOCD_BUILD_ARGS} --image-builder ${ARGOCD_OPERATOR_IMAGE_BUILDER}
```

Comment:
Should we document what ARGOCD_BUILD_ARGS is for and what can be passed?

Reply:
You can pass anything to the operator-sdk you like.

Reply:
But there is no words on the intend to pass "openshift".

Reply:
Yeah, the whole point of using build tags is that we don't _need_ to and by default the openshift specific bits won't be included?

We could probably even remove the openshift.go from the cmd package (which uses the build tags) and leave it to the gitops-operator to do the registration (by importing).

Reply:
I like the latter.  Let's leave the OpenShift specific to `gitops-operator` when possible.

Reply:
This is complicated tho', we do want to shift the openshift routing out of the core reconciliation.

Reply:
I added a documentation(comment) which explains the behaviour of ARGOCD_BUILD_ARGS. I will resolve the conversation if it looks good.